gui-agent/vmside.c

```diff
@@ -2279,6 +2320,71 @@ int main(int argc, char **argv)
     int xfd;
     Ghandles g;
     int wait_fds[2];
+    
+        
+    // open uinput, if it fails, falls back to xdriver to not break input to the qube
+    g.output_fd = open("/dev/uinput", O_WRONLY | O_NDELAY);
```

Comment:
Please make this feature opt-in, not the default. You can use qvm-service framework for that - see how clipboard wipe is handled for example.

Reply:
Would it be alright for it to be on by default in whonix workstation qubes, and then opt-in for all other qubes? If not, it's no problem and I can make it opt-in for all.

Reply:
There are ways to enable "qvm-service" in whonix by default, either via https://github.com/QubesOS/qubes-core-admin-addon-whonix/, or by simply touching relevant file in `ExecStartPre=` added with a service drop-in file.

Reply:
Sorry didn't realize this needed to be marked as resolved. I made it opt-in using a qvm-service with the name gui-agent-virtual-input-device in commit c2f6757c65247c63c8e1e5889dc324027bee0f2c